src/main/java/frc/robot/swerve/ModuleIOReal.java

```diff
@@ -0,0 +1,165 @@
+package frc.robot.swerve;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.*;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import frc.robot.Constants;
+import utils.math.AngleUtil;
+import utils.math.differential.Integral;
+
+public class ModuleIOReal implements ModuleIO {
+
+    private final TalonFX driveMotor;
+    private final TalonFX angleMotor;
+    private final DutyCycleEncoder encoder;
+
+    private final Integral driveSupplyChargeUsedCoulomb = new Integral();
+    private final Integral driveStatorChargeUsedCoulomb = new Integral();
+    private final Integral angleSupplyChargeUsedCoulomb = new Integral();
+    private final Integral angleStatorChargeUsedCoulomb = new Integral();
+    private Rotation2d angleSetpoint;
+    private Rotation2d currentAngle;
+    private double driveMotorVelocitySetpoint;
+
+    private MotionMagicVoltage angleControlRequest = new MotionMagicVoltage(0).withEnableFOC(true);
+    private VelocityVoltage velocityControlRequest = new VelocityVoltage(0).withEnableFOC(true);
+
+    public ModuleIOReal(int driveMotorID, int angleMotorID, int encoderID, int number,
+                        TalonFXConfiguration driveConfig, TalonFXConfiguration angleConfig) {
+        this.driveMotor = new TalonFX(driveMotorID);
+        this.angleMotor = new TalonFX(angleMotorID);
+
+        this.encoder = new DutyCycleEncoder(encoderID);
+
+        TalonFXConfiguration driveMotorConfig = new TalonFXConfiguration();
+        driveMotorConfig = driveConfig;
+        driveMotorConfig.ClosedLoopGeneral.ContinuousWrap = true;
```

Comment:
Why is that needed here?

Reply:
`driveMotorConfig` really isn't needed.

Reply:
Why not? They are needed in general, I was asking specifically about setting continuous wrap on the drive motor.

Reply:
They are needed but not in a separate variable, they're only used once.
Continuous wrap doesn't have a `.withContinuousWrap()` method so it can't be added to the config object in the constants.

Reply:
Ok, but what's preventing you to run DRIVE_MOTOR_CONFIGS.ClosedLoopGeneral.ContinuousWrap = true;` in the constants file?
However, my original question was why do you need to configure continuous wrap on a motor with no "start" and "end position" like the angle motor.

Reply:
> what's preventing you to run DRIVE_MOTOR_CONFIGS.ClosedLoopGeneral.ContinuousWrap = true; in the constants file?

I've tried, it says it's an unknown class.

>why do you need to configure continuous wrap on a motor with no "start" and "end position"

Yeah you're right I only need it for the angle motor.

Reply:
Can you attach a screenshot of the line of code and the error?

Reply:
![image](https://github.com/Galaxia5987/common/assets/111113203/adfc69ca-ebe7-44ab-8af2-e744d3c5ccf9)

Reply:
Oh, I see. Forgot Java has this thing.
You have two options here:
1. Use a [`static` block](https://stackoverflow.com/a/2943575/10015776) around these lines. However, in this case, it's better to just wrap everything in a `static` block, which might be a bit weird.
2. Leave this line in the subsystem as it is now.

The choice is yours🙂